src/dune_pkg/rev_store.ml

```diff
@@ -53,24 +52,20 @@ let show { dir; lock = _ } (Rev rev) path =
 
 let create ~dir =
   let lock = Fiber.Mutex.create () in
-  { dir; lock }
-;;
-
-let initialize ({ dir; lock } as t) =
-  Fiber.Mutex.with_lock lock ~f:(fun () ->
-    let* () = Fiber.return () in
-    let+ () =
-      match Fpath.mkdir_p (Path.to_string dir) with
-      | Already_exists -> Fiber.return ()
-      | Created -> run t [ "init"; "--bare" ]
-      | exception Unix.Unix_error (e, x, y) ->
-        User_error.raise
-          [ Pp.textf "%s isn't a directory" (Path.to_string_maybe_quoted dir)
-          ; Pp.textf "reason: %s" (Unix_error.Detailed.to_string_hum (e, x, y))
-          ]
-          ~hints:[ Pp.text "delete this file or check its permissions" ]
-    in
-    t)
+  let t = { dir; lock } in
+  let* () = Fiber.return () in
+  let+ () =
+    match Fpath.mkdir_p (Path.to_string dir) with
+    | Already_exists -> Fiber.return ()
+    | Created -> run t [ "init"; "--bare" ]
+    | exception Unix.Unix_error (e, x, y) ->
+      User_error.raise
+        [ Pp.textf "%s isn't a directory" (Path.to_string_maybe_quoted dir)
+        ; Pp.textf "reason: %s" (Unix_error.Detailed.to_string_hum (e, x, y))
```

Comment:
```suggestion
    | exception Unix.Unix_error unix_err ->
      User_error.raise
        [ Pp.textf "%s isn't a directory" (Path.to_string_maybe_quoted dir)
        ; Pp.textf "reason: %s" (Unix_error.Detailed.to_string_hum unix_err)
```

Reply:
This seems to be one of the cases where constructors of arity 3 and 3-tuples are not treated the same, so unfortunately it has to be reconstructed.

Reply:
I'm trying to think how this can even occur. Would `Fpath.mkdir_p` return `Already_exists`? Have a look at `mkdir_p` inside `install_uninstall.ml`. There is even a change request for `Fpath` to handle this case better, and I even tried to solve it previously https://github.com/ocaml/dune/pull/6355 but it got reverted due to a perf bug.